node/src/server-modules/GlideJson.ts

```diff
@@ -25,6 +25,19 @@ export interface JsonGetOptions {
     noescape?: boolean;
 }
 
+/** Additional options for {@link GlideJson.arrpop | JSON.ARRPOP} command. */
+export type JsonArrPopOptions =
+    | {
+          /** The path within the JSON document. */
+          path?: GlideString;
```

Comment:
does this work? type union will just combine the types together and make path optional in both cases. 
you can make it work with different field names